zebra-network/src/peer_set/set.rs

```diff
@@ -172,6 +184,10 @@ where
         }
     }
 
+    /// Check background task handles to make sure they're still running.
```

Comment:
Question: Why aren't these background tasks handled separately? Wouldn't it make more sense to have this service have a `Request::Stop` or a `cancel_signal: oneshot::Receiver<()>`?

Reply:
I think the shutdown refactor is part of another(already merged) PR and the related commit should not be here:
https://github.com/ZcashFoundation/zebra/pull/3071/commits

Reply:
Yep, sorry about that, but I'll leave this PR open so I remember to deal with the tasks in PeerSet's Drop.

Reply:
Answered in PR #3078.